pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -635,6 +635,24 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 	return changes
 }
 
+// We will continue killing containers in the case of error return from killContainer
+func (m *kubeGenericRuntimeManager) killAllContainersIgnoringFailures(pod *v1.Pod, podContainerChanges podActions, result kubecontainer.PodSyncResult) (int, bool) {
```

Comment:
Seems like the first `int` return value is only used for testing.  I'm not sure we want to add return values that have no function in the program.  Seems like a unit test anti-pattern.

This makes me look at the test and realize we don't want to expose the function's internal counter, but rather, check to make sure that `killContainer()` was called for all `ContainersToKill`, even if `killContainer` fails for one of them.

I'd have to look deeper to see how this could be done.

Reply:
yeah, we should use the `FakeRuntimeService` returned by `createTestRuntimeManager`, load the mocked containers (say three containers, with one container that has some condition that makes `killContainer` fail), and ensure that `StopContainer` was called for each container where `killContainer` was expected to succeed.

Reply:
I have dropped the first return value and modified the test.
Please take a look.

Reply:
@sjenning 
Please take another look.